crates/hulk/src/nao/double_buffered_reader.rs

```diff
@@ -0,0 +1,513 @@
+use std::{
+    io::{self, ErrorKind, Read},
+    mem::{size_of, MaybeUninit},
+    os::unix::{io::AsRawFd, prelude::RawFd},
+    ptr::null_mut,
+};
+
+use libc::{fd_set, select, FD_SET, FD_ZERO};
+
+use super::hula::StateStorage;
+
+const NUMBER_OF_BUFFERS: usize = 2;
+
+pub struct DoubleBufferedReader<Reader, Poller> {
+    reader: Reader,
+    poller: Poller,
```

Comment:
Why is it generic over the `poller`?

Reply:
For dependency injection for testability.